pySDC/implementations/controller_classes/controller_nonMPI.py

```diff
@@ -74,23 +74,12 @@ def __init__(self, num_procs, controller_params, description):
             self.logger.warning('you have specified a predictor type but only a single level.. '
                                 'predictor will be ignored')
 
-        # prepare variables to do with error estimation and resilience
-        self.params.use_embedded_estimate = self.params.use_embedded_estimate or self.params.use_adaptivity or\
-            self.params.use_HotRod
-        self.params.use_extrapolation_estimate = self.params.use_extrapolation_estimate or self.params.use_HotRod
-        self.store_uold = self.params.use_iteration_estimator or self.params.use_embedded_estimate
-        if self.params.use_adaptivity:
-            if 'e_tol' not in description['level_params'].keys():
-                raise ParameterError('Please supply "e_tol" in the level parameters')
-            if 'restol' in description['level_params'].keys():
-                if description['level_params']['restol'] > 0:
-                    description['level_params']['restol'] = 0
-                    self.logger.warning(f'I want to do always maxiter={description["step_params"]["maxiter"]} iteration\
-s to have a constant order in time for adaptivity. Setting restol=0')
-        if self.params.use_HotRod and self.params.HotRod_tol == np.inf:
-            self.logger.warning('Hot Rod needs a detection threshold, which is now set to infinity, such that a restart\
- is never triggered!')
-        self.error_estimator = get_ErrorEstimator_nonMPI(self)
+        self.store_uold = self.params.use_iteration_estimator
+
+        self.add_convergence_controller(BasicRestartingNonMPI, {}, description)
+
```

Comment:
I don't really see the advantage of a `None` instead of an empty dict here. That just adds more lines in your other comment. Can you elaborate on why this is better?

Reply:
[Dangerous business](https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python) and less explicit than keyword arguments.